shophomework/src/ro/siit/PurchaseManager.java

```diff
@@ -0,0 +1,52 @@
+package ro.siit;
+
+import java.text.ParseException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+
+import static ro.siit.Customer.AGE_RESTRICTION_ADULT;
+import static ro.siit.Customer.AGE_RESTRICTION_TEENAGER;
+
+
+public class PurchaseManager {
+
+    public static void processPurchase(Product product, Customer customer) throws ParseException {
```

Comment:
Daca se poate face tranzactia trebuie sa scazi balance-ul customerului si quantity-ul produsului. Daca una dintre conditii nu este indeplinita atunci trebuie sa iesi din metoda fara a face tranzactia (se poate iesi din metoda folosind return). In cazul tau dupa ce evaluezi if (product.getQuantity() < 1)  afisezi doar un mesaj dar metoda merge mai departe.